core/src/test/java/org/apache/iceberg/TestMetadataUpdateParser.java

```diff
@@ -108,19 +121,19 @@ public void testUpgradeFormatVersionFromJson() {
   }
 
   /** AddSchema * */
-  @Test
-  public void testAddSchemaFromJson() {
+  @ParameterizedTest
+  @MethodSource("testSchemas")
```

Comment:
This also abuses existing test cases to test different behavior than was intended. I think tests should be written directly instead.

Reply:
I reverted this change and have a new test to cover that.